controllers/openstackcluster_controller.go

```diff
@@ -395,7 +395,11 @@ func (r *OpenStackClusterReconciler) reconcileBastion(ctx context.Context, scope
 		return nil, err
 	}
 
-	return bastionAddFloatingIP(openStackCluster, clusterResourceName, port, networkingService)
+	if !ptr.Deref(openStackCluster.Spec.DisableExternalNetwork, false) {
+		return bastionAddFloatingIP(openStackCluster, clusterResourceName, port, networkingService)
+	}
+
+	return nil, nil
```

Comment:
What do you think about doing this the other way? If disabled -> log something and return. Otherwise continue and add it.

Reply:
IIUC what you're proposing is

```golang
...
if ptr.Deref(openStackCluster.Spec.DisableExternalNetwork, false) {
        // log something? what exactly?
	return nil, nil
}
return bastionAddFloatingIP(openStackCluster, clusterResourceName, port, networkingService)
```

If yes, I think it's functionally the same and is fairly consistent with other Deref calls for that field. No strong opinion here, happy to make the change.